### ISSUE TITLE
Help needed: username change

### ISSUE THREAD

**Issue by Shark-93** [andersonkrs/malheatmap#965]
Hey there, me again :)

I've submitted a special one-time request to the MAL administrator, to change my username with one that was already claimed by another user, but this should not be important; basically I had the admins change my username and since then I'm afraid MAL Heatmap has stopped tracking me.

The Heatmap page still shows my old username on it as well as in the URL user/S_h_a_r_k_93, but my new username is Shark93. Are you able to update my MAL Heatmap username and therefore fix the sync between my "new" MAL profile and the Heatmap profile?

Thank you very much, and sorry for the inconvenience! I've noticed the problem only today; I don't mind losing last week of May history updates, it's okay.

**Comment by andersonkrs** [comment]
Sup @Shark-93 !

Have you tried to click "Sign out" and then logging back with your MAL account? 

In theory, your account S_h_a_r_k_93 is associated with a MAL ID. When logging in back we sync some profile data like the username, so your user would be renamed. 🤔 Here on the DB says the last time we synched your profile data was in April 🤔

**Comment by Shark-93** [comment]
Hola! April is when I first signed up to MAL Heatmap, with my previous username; then end of May is when I had the username update by the Staff (hopefully equivalent to a normal username update).

I had actually also read your **FAQ** after opening this issue,  and this scenario is actually covered, but in a rather pessimistic way 😅 maybe we can take this opportunity to update the documentation at once.

Basically my heatmap page was a bit broken since May: no updates and **no sign out / profile data** was shown on the left hand side; the old username was shown and clicking on it brought me to a _now invalid_ MAL profile.

What I did to sort of re-sync is clicking on the MAL Heatmap logo at the top of the page, which prompted me to log in via a myanimelist button: doing that prompted me to allow MAL Heatmap again, but this time my new username was shown.

Giving permission to the external app to access my MAL data, I finally had a semi-working heatmap page, with my correct profile info on the left, Sign out button, updated username, but still no history updates past May 28th.

The stats on the left read _Updated at 17 Jun 2024 22:25 +0200_ but I am unsure if it references only the profile data, or the history; or if the update timestamp is just placebo. 😝

When you read this issue in the next days, please check my profile again: I hope that the daily syncing routine has caught up with my past 3 weeks of MAL history updates, and therefore that the sync has also been re-established.

That's what I expect, at least, now that I have signed back in. Ah, final footnote, contrary to what the FAQ says I didn't actually lose my data in the process; like you said in this issue you perhaps TRACK HISTORY by username, but you IDENTIFY USER by user id so... I have lost tracking for a little bit, but I haven't lost user data.

The FAQ should be updated, maybe by performing some more tests / following my steps described above.

**TL;DR**
heatmap profile auto-fixed
heatmap sync to be checked/fixed
heatmap faq to be checked/updated

**Comment by andersonkrs** [comment]
Hey @Shark-93 I've updated the FAQ. Indeed it was stale. Before we didn't have MAL OAuth sign in. So the two accounts could be disconnected at some point if the user changed their username.

Also the synching caught up.

**Comment by Shark-93** [comment]
Thank you. Can you please check why it is not syncing anymore on my profile, since the 27th? Is it just me or a global problem? Your profile doesn't fully load either. Thanks!
